src/dev/flang/be/c/C.java

```diff
@@ -446,32 +447,33 @@ else if (!_fuir.clazzIsVoidType(tc))
                   ctags.add(CExpr.int32const(tagNum).comment(_fuir.clazzAsString(tc)));
                   if (CHECKS) check
                     (hasTag || !_fuir.hasData(tc));
+                  cTagNum = tagNum;
                 }
             }
           if (tags.length > 0)
-             {
-               var sl = new List<CStmnt>();
-               var field = _fuir.matchCaseField(s, mc);
-               if (field != NO_CLAZZ)
-                 {
-                   var fclazz = _fuir.clazzResultClazz(field);     // static clazz of assigned field
-                   var cl     = _fuir.clazzAt(s);
-                   var f      = field(cl, C.this.current(s), field);
-                   var entry  = _fuir.clazzIsRef(fclazz) ? ref.castTo(_types.clazz(fclazz)) :
-                                _fuir.hasData(fclazz)   ? uniyon.field(new CIdent(CNames.CHOICE_ENTRY_NAME + tags[0]))
-                                                         : CExpr.UNIT;
-                   sl.add(C.this.assign(f, entry, fclazz));
-                 }
-               sl.add(ai.processCode(_fuir.matchCaseCode(s, mc)).v1());
-               sl.add(CStmnt.BREAK);
-               var cazecode = CStmnt.seq(sl);
-               tcases.add(CStmnt.caze(ctags, cazecode));  // tricky: this a NOP if ctags.isEmpty
-               if (!rtags.isEmpty()) // we need default clause to handle refs without a tag
-                 {
-                   rcases.add(CStmnt.caze(rtags, cazecode));
-                   tdefault = cazecode;
-                 }
-             }
+            {
+              var sl = new List<CStmnt>();
+              var field = _fuir.matchCaseField(s, mc);
+              if (field != NO_CLAZZ)
+                {
+                  var fclazz = _fuir.clazzResultClazz(field);     // static clazz of assigned field
+                  var cl     = _fuir.clazzAt(s);
+                  var f      = field(cl, C.this.current(s), field);
+                  var entry  = _fuir.clazzIsRef(fclazz) ? ref.castTo(_types.clazz(fclazz)) :
+                               _fuir.hasData(fclazz)    ? uniyon.field(CIdent.choiceEntry(cTagNum))
+                                                        : CExpr.UNIT;
+                  sl.add(C.this.assign(f, entry, fclazz));
+                }
```

Comment:
```suggestion
                }
              var ctags = ctagNums.map(i -> CExpr.int32const(i).comment(_fuir.clazzAsString(_fuir.clazzChoice(subjClazz,i)));
```
our put this code directly in the line `tcases.add(CStmnt.caz(ctags,...` below, I cannot add a suggestions there.